### PR TITLE
IPQ 05: Network Delay Time

### PR DESCRIPTION
**Pull request by char-adadev**
The purpose of this PR is to introduce the Network Delay Time IPQ. This PR builds on the PR for Bipartition Graph.

For review:
* Please run learn preview
* Verify the example implementation passes tests
* Tests are new so please let me know if anything looks amiss there!
* Any suggestions for wording, grammar, etc are welcome

### CODE REVIEW

File: 05/network-delay.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
+# Network Delay
+
+## Problem Statement  
+
+For this exercise, create a function `network_delay_time` which takes in a list of travel times, `times`, as directed edges `times[i] = (uᵢ, vᵢ, wᵢ)` where `uᵢ` is the source node, `vᵢ` is the target node, and `wᵢ` is the time it takes for a signal to travel from the source node to the target node, `n`, the total number of nodes in the graph, and `source`, the node from which a signal is being sent. The nodes in the graph are labeled from `1` to `n`.
+
+Return the **minimum** time it takes for all of the nodes to receive the signal from the `source` node. If it is not possible for all of the nodes to receive the signal, return `-1`.
```
Comment (kyra-at-ada) [inline 05/network-delay.md:7]:
```suggestion
Return the **minimum** time it takes for **all** of the nodes in the graph to receive the signal from the `source` node. If it is not possible for all of the nodes to receive the signal, return `-1`.
```

File: 05/network-delay.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    source = 1
+
+    # Act
+    answer = network_delay_time(times, n, source)
+
+    # Assert
+    self.assertEqual(answer, 9)
+```
+
+</details>
```
Comment (kyra-at-ada) [inline 05/network-delay.md:116]: Consider keeping tests just in final challenge
```suggestion
```

File: 05/implement-network-delay.instructor.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Instructor: Implement Network Delay
+
+An example of a working implementation:
+
+```python
+import collections
+from heapq import heappush, heappop
+def network_delay_time(times, n, source):
+    # Initialize dictionary with default value as a list
```
Comment (kyra-at-ada) [inline 05/implement-network-delay.instructor.md:9]: Folks got so hung up on default dict last cohort. Should we link them to the documentation?

**Review by kyra-at-ada — APPROVED** [review]
Looks good! Added some minor suggestions. 

- Do we want to add in a question that asks them to consider what graph representation they are given in this problem? I was thinking it might be a good way to indicate one of the extra things they need to think about when presented a graph problem.

**Comment by char-adadev** [comment]
> Looks good! Added some minor suggestions.
> 
> * Do we want to add in a question that asks them to consider what graph representation they are given in this problem? I was thinking it might be a good way to indicate one of the extra things they need to think about when presented a graph problem.

Good shout out! There's currently an example clarifying question (in the explanation for listing more questions) which asks "In which way is the graph being represented?". Do you think this is a good spot?

**Pull request merged**